clippy_lints/src/types.rs

```diff
@@ -236,6 +263,24 @@ fn match_type_parameter(cx: &LateContext<'_, '_>, qpath: &QPath<'_>, path: &[&st
     false
 }
 
+/// Checks if `qpath` last segment type parameter is a reference
+fn match_type_parameter_rptr(qpath: &QPath<'_>) -> bool {
+    let last = last_path_segment(qpath);
+    if_chain! {
+        if let Some(ref params) = last.args;
+        if !params.parenthesized;
+        if let Some(ty) = params.args.iter().find_map(|arg| match arg {
+            GenericArg::Type(ty) => Some(ty),
+            _ => None,
+        });
```

Comment:
Oh this lint was meant to lint all types `Box<&_>`, not only on generics borrowed in a box. The `T` is often chosen as a wildcard for types, like `_` is chosen as a wildcard for names, when describing an issue or similar things.

Reply:
thanks! As far as I understand it lints both generics and concrete types, but the error message was poor with hardcoded `&T`. Trying to make it better.

Reply:
Hm, then I maybe misunderstood the code.